test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -261,6 +270,8 @@ private static void validateSpec(RestSpec restSpec) {
 
     @After
     public void wipeCluster() throws Exception {
+        // Running tasks might create indices and otherwise invalidate the wipe. Lets just let the finish.
+        waitForAllRunningTasksToFinish();
```

Comment:
> First is a user deleting an index and it causing problems.

It'll cause the reindex request to fail, logging a warning. That seems fine.

> cancel all running tasks

I can try to cancel the tasks if you'd like but they don't always listen to a cancel request and cancellation is async. Reindex only checks the cancel flag in between requests, for example. If I cancel it and it's already started an index request that will create a new index then the index will still be created. 

So I can cancel all the running tasks but I'd still have to check to see that they've killed themselves with the same timeout. I'd just have a different exception message - instead of "timed out waiting for running tasks" it'd be "timed out waiting for tasks to die".